client/my-sites/themes/auto-loading-homepage-modal.jsx

```diff
@@ -49,11 +51,41 @@ class AutoLoadingHomepageModal extends Component {
 		installingThemeId: PropTypes.string.isRequired,
 	};
 
+	constructor( props ) {
+		super( props );
+		this.state = {
+			homepageAction: 'keep_current_homepage',
+
+			// Used to reset state when dialog re-opens
+			wasVisible: props.isVisible,
+		};
+	}
+
+	static getDerivedStateFromProps( nextProps, prevState ) {
+		if ( nextProps.isVisible && ! prevState.wasVisible ) {
+			return { homepageAction: 'keep_current_homepage', wasVisible: true };
+		} else if ( ! nextProps.isVisible && prevState.wasVisible ) {
+			return { wasVisible: false };
+		}
+		return null;
+	}
```

Comment:
Using state derived from props is a bit messy, but I want to reset the state each time the dialog opens so that `keep_current_homepage` is always selected by default.

_Ideally_ I'd wrap the radio buttons in a `<form>` and grab the current selection when the user submits the form. That way we wouldn't need to maintain any state in the component. It'd be better for a11y too.
Unfortunately the interface for `<Dialog>` doesn't have a way to let me place the dialog buttons inside the `<form>` and make one of them `type="submit"`.

Reply:
Agreed, it's a bit hacky but probably the best way to reset the state given the limitations from `<Dialog />`.

Maybe it would be worth to add a comment inside the `getDerivedStateFromProps` function, explaining that the code in there makes sure that `keep_current_homepage` is always the default selection when the dialog gets opened?